src/WebJobs.Extensions.Twilio/Bindings/TwilioScriptBindingProvider.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Reflection;
+using Microsoft.Azure.WebJobs.Extensions.Twilio;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.Extensibility;
+using Newtonsoft.Json.Linq;
+using Twilio;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Bindings
+{
+    /// <summary>
+    /// BindingProvider for Twilio extensions
+    /// </summary>
+    public class TwilioScriptBindingProvider : ScriptBindingProvider
+    {
+        private readonly string _twilioAssemblyName;
+
+        /// <inheritdoc/>
+        public TwilioScriptBindingProvider(JobHostConfiguration config, JObject hostMetadata, TraceWriter traceWriter) 
+            : base(config, hostMetadata, traceWriter)
+        {
+            _twilioAssemblyName = typeof(SMSMessage).Assembly.GetName().Name;
+        }
+
+        internal TwilioSMSConfiguration CreateConfiguration(JObject metadata)
+        {
+            var twilioConfig = new TwilioSMSConfiguration();
+
+            JObject configSection = (JObject)metadata.GetValue("twilioSMS", StringComparison.OrdinalIgnoreCase);
```

Comment:
Yes, that was the thought. I do want to support other Twilio services (e.g. voice, tasks, etc.), that's why I qualified this.